CondFormats/HcalObjects/src/HcalCondObjectContainerBase.cc

```diff
@@ -13,9 +13,9 @@ HcalCondObjectContainerBase::HcalCondObjectContainerBase(const HcalTopology* top
 }
 
 void HcalCondObjectContainerBase::setTopo(const HcalTopology* topo) {
-  if (topo && !topo->denseIdConsistent(packedIndexVersion_)) {
+  if ((topo) && (packedIndexVersion_ != 0) &&

```

Comment:
The default constructor initializes packedIndexVersion_ = 0
It does not know at that stage what the packed index version is. Ideally the default constructor should know the packed index version at the time of construction. Unfortunately the packed index version is not saved in the data base (except the geometry in DB). We may at some point in time utilize that. It will come later on. At the moment the code is unprotected and this warning is totally misleading
________________________________
From: Marco Musich [notifications@github.com]
Sent: 05 April 2016 14:37
To: cms-sw/cmssw
Cc: Sunanda Banerjee
Subject: Re: [cms-sw/cmssw] bsunanda:Run2-hcx76 Remove wrong warnings (#13915)


In CondFormats/HcalObjects/src/HcalCondObjectContainerBase.cc<https://github.com/cms-sw/cmssw/pull/13915#discussion_r58529923>:

> @@ -13,9 +13,9 @@ HcalCondObjectContainerBase::HcalCondObjectContainerBase(const HcalTopology* top
>  }
>
>  void HcalCondObjectContainerBase::setTopo(const HcalTopology* topo) {
> -  if (topo && !topo->denseIdConsistent(packedIndexVersion_)) {
> +  if ((topo) && (packedIndexVersion_ != 0) &&


@bsunanda<https://github.com/bsunanda> which use case (packedIndexVersion_ != 0) is supposed to take care of?

—
You are receiving this because you were mentioned.
Reply to this email directly or view it on GitHub<https://github.com/cms-sw/cmssw/pull/13915/files/8b8909957f8e7e2d99992087483b106ff7e2b6b8#r58529923>